### ISSUE TITLE
Calling "make" in a loop produces invalid QR codes.

### ISSUE THREAD

**Issue by jtmcdole** [kevmoo/qr.dart#45]
Setup:
1. Create a QR code with some data.
2. Loop mask 0 to 7: call make(mask)
3. render qr code

Expected:
Images with different masks

Actual:
Images with same data and format bars change.

![foo_0](https://user-images.githubusercontent.com/1924313/123035952-1c203400-d3a1-11eb-9efd-2992d5e35811.png)
![foo_1](https://user-images.githubusercontent.com/1924313/123035954-1c203400-d3a1-11eb-9d35-36eab7e2ede1.png)
![foo_2](https://user-images.githubusercontent.com/1924313/123035955-1cb8ca80-d3a1-11eb-8e29-f0ab27004e7b.png)
![foo_3](https://user-images.githubusercontent.com/1924313/123035957-1cb8ca80-d3a1-11eb-897a-0f626d4dd292.png)
![foo_4](https://user-images.githubusercontent.com/1924313/123035962-1cb8ca80-d3a1-11eb-9a46-427d1f2fe34b.png)
![foo_5](https://user-images.githubusercontent.com/1924313/123035964-1cb8ca80-d3a1-11eb-9773-a4eaaf5659a5.png)
![foo_6](https://user-images.githubusercontent.com/1924313/123035965-1d516100-d3a1-11eb-81bd-824ba70df97f.png)
![foo_7](https://user-images.githubusercontent.com/1924313/123035966-1d516100-d3a1-11eb-99af-e4b0bba391b7.png)

**Comment by jtmcdole** [comment]
Expected results gained by creating new QR code each time with the same data:

![foo_0](https://user-images.githubusercontent.com/1924313/123036069-4b36a580-d3a1-11eb-85ae-e5c471c5a88e.png)
![foo_1](https://user-images.githubusercontent.com/1924313/123036070-4b36a580-d3a1-11eb-823e-6dfc39cc9729.png)
![foo_2](https://user-images.githubusercontent.com/1924313/123036071-4bcf3c00-d3a1-11eb-9d53-734bb3313453.png)
![foo_3](https://user-images.githubusercontent.com/1924313/123036072-4bcf3c00-d3a1-11eb-87db-0db94c6cbf13.png)
![foo_4](https://user-images.githubusercontent.com/1924313/123036073-4bcf3c00-d3a1-11eb-9b22-6a259194a7ea.png)
![foo_5](https://user-images.githubusercontent.com/1924313/123036074-4bcf3c00-d3a1-11eb-969d-7d16cf2bcaec.png)
![foo_6](https://user-images.githubusercontent.com/1924313/123036075-4c67d280-d3a1-11eb-852f-b7a7fc4c5700.png)
![foo_7](https://user-images.githubusercontent.com/1924313/123036076-4c67d280-d3a1-11eb-9e83-4532d2a0395b.png)

**Comment by kevmoo** [comment]
This code is VERY old. I basically copied a JS impl. PRs welcome to fix this type of issue!

**Comment by jtmcdole** [comment]
I'm happy to look into it.

**Comment by kevmoo** [comment]
I'm wondering if `make` should return the bytes. And we don't store in in `QrCode`

**Comment by jtmcdole** [comment]
That sounds like a good separation of concerns to me.  I'm about to return from lunch; so I can look at it tonight.

**Comment by jtmcdole** [comment]
So its not `_dataCache`, its `_modules` and I think your suggestion would work.

This would be a breaking change though, since `qrCode.isDark(y,x)` would no longer point at these bytes... _or_ we keep the bytes, mark as deprecated, and make a copy returned from make().

I'm OK with the following, which is basically "just re-init the modules".

```
  void make([int? maskPattern]) {
    assert(maskPattern == null || (maskPattern >= 0 && maskPattern <= 7));
    _modules.clear();
    for (var row = 0; row < moduleCount; row++) {
      _modules.add(List<bool?>.filled(moduleCount, null));
    }
    _makeImpl(false, maskPattern ?? _getBestMaskPattern());
  }
```

**Comment by jtmcdole** [comment]
Compromise option:

1. modules comes "out" of qr code.
2. make() finalizes a new {-QrData / QrModule / QrImage}?
3. We hold on to the last one made by make()
4. isDark() goes to this new class.

**Comment by jtmcdole** [comment]
Fixed by #48
